src/data/nat/factorization.lean

```diff
@@ -119,6 +119,35 @@ begin
     simp [prod_insert hxT, sum_insert hxT, ←IH, factorization_mul (hS x hxS) hT] }
 end
 
+/-! ### Bijection between pnats and finsupps `ℕ →₀ ℕ` with support on the primes -/
+
+/-- Any finsupp `f : ℕ →₀ ℕ` whose support is in the primes is equal to the factorization of
+the product `∏ (a : ℕ) in f.support, a ^ f a`. -/
+lemma factorization_prod_pow_inv {f : ℕ →₀ ℕ} (hf : ∀ (p : ℕ), p ∈ f.support → prime p) :
+  (f.prod pow).factorization = f :=
+begin
+  have h : ∀ x : ℕ, x ∈ f.support → x ^ f x ≠ 0 := λ p hp, pow_ne_zero _ (prime.ne_zero (hf p hp)),
+  simp only [finsupp.prod, factorization_prod h],
+  nth_rewrite_rhs 0 (sum_single f).symm,
+  exact sum_congr rfl (λ p hp, prime.factorization_pow (hf p hp)),
+end
+
+lemma prime_finsupp_prod_pow_pos {f : ℕ →₀ ℕ} (hf : ∀ p ∈ f.support, prime p) : 0 < f.prod pow :=
```

Comment:
I get this is a convenience method but maybe it should be `hf : ¬ 0 ∈ f.support` instead and be somewhere a bit more general?

Reply:
I've replaced this with 
```
lemma prod_pow_pos_of_ne_zero {f : ℕ →₀ ℕ} (hf : 0 ∉ f.support) : 0 < f.prod pow
```
but I'm not sure where else to put it — it needs to go somewhere that can see `finset.prod_pos`, which is defined in `algebra/big_operators/order`, which doesn't know about `finsupp`.

Reply:
does `algebra/big_operators/finsupp` not work?

Reply:
It seems not, `finset.prod_pos` isn't recognised there.

Reply:
I dob't think importing the file is too bad there

Reply:
Ok, I've moved it to there and added the import of `algebra/big_operators/finsupp` to `data/nat/factorization`.